### PR TITLE
Update CONTRIBUTING.md to reflect the necessary steps

### PR DESCRIPTION
**Pull request by DenisGorbachev**
**Summary**

This PR allows potential contributors to get started faster.

**Test plan**

I ran these commands myself to setup a new fork for contributing.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -8,9 +8,12 @@ please read the [code of conduct](CODE_OF_CONDUCT.md).
 > Install yarn on your system: https://yarnpkg.com/en/docs/install
 
 ```sh
-$ git clone https://github.com/yarnpkg/yarn.git
+$ # fork the repo
+$ git clone YOUR_YARN_REPO_URL
 $ cd yarn
-$ yarn
+$ npm install
```
Comment (bestander) [inline CONTRIBUTING.md]: contributing to yarn does not require npm.
You can have yarn installed globally to get node_modules

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,9 +8,12 @@ please read the [code of conduct](CODE_OF_CONDUCT.md).
 > Install yarn on your system: https://yarnpkg.com/en/docs/install
 
 ```sh
-$ git clone https://github.com/yarnpkg/yarn.git
+$ # fork the repo
+$ git clone YOUR_YARN_REPO_URL
```
Comment (bestander) [inline CONTRIBUTING.md]: YOUR_YARN_REPO_URL is more confusing, it implies that a person forked yarn on github

Comment (DenisGorbachev) [inline CONTRIBUTING.md]: That's right - the instructions in `CONTRIBUTING.md` indicate that the contributor should fork the repo.

**Comment by DenisGorbachev** [comment]
@bestander applied the fixes as you've requested - how about now? 

@bestander what about first-time contributors: is it better for them to follow clear instructions or to figure them out?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -5,12 +5,13 @@ please read the [code of conduct](CODE_OF_CONDUCT.md).
 
 ## Setup
 
-> Install yarn on your system: https://yarnpkg.com/en/docs/install
-
 ```sh
-$ git clone https://github.com/yarnpkg/yarn.git
+$ # Install yarn on your system: https://yarnpkg.com/en/docs/install
```
Comment (bestander) [inline CONTRIBUTING.md]: The previous formatting was better, the link was clickable.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -5,12 +5,13 @@ please read the [code of conduct](CODE_OF_CONDUCT.md).
...
+$ # Fork the repo: https://github.com/yarnpkg/yarn
```
Comment (bestander) [inline CONTRIBUTING.md]: Same thing for this instruction, I don't think it should be in a comment of bash script.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -5,12 +5,13 @@ please read the [code of conduct](CODE_OF_CONDUCT.md).
...
+$ git clone YOUR_YARN_REPO_URL
```
Comment (bestander) [inline CONTRIBUTING.md]: YOUR_YARN_REPO_URL is confusing
```
git clone  https://github.com/<YOUR_GITHUB_NAME>/yarn yarn
```

**Comment by DenisGorbachev** [comment]
@bestander Is this better now?

On an extra note, I've also replaced the mentions of `npm` with `yarn`

**Comment by bestander** [comment]
Thanks, @DenisGorbachev

**Comment by DenisGorbachev** [comment]
Thank you too, Konstantin!

**Pull request merged**